packages/components/src/DataList/DataList.tsx

```diff
@@ -18,32 +24,49 @@ import {
   DataListItems,
 } from "./components/DataListLayoutInternal";
 import { useLayoutMediaQueries } from "./hooks/useLayoutMediaQueries";
-import { DataListContext } from "./context/DataListContext";
+import { DataListContext, useDataListContext } from "./context/DataListContext";
 import {
   DataListFilters,
   InternalDataListFilters,
 } from "./components/DataListFilters";
+import {
+  DataListSearch,
+  InternalDataListSearch,
+} from "./components/DataListSearch";
 import { Heading } from "../Heading";
 
 export function DataList<T extends DataListObject>(props: DataListProps<T>) {
+  const searchComponent = getCompoundComponent<DataListSearchProps>(
+    props.children,
+    DataListSearch,
+  );
+  const filterComponent = getCompoundComponent<DataListFiltersProps>(
+    props.children,
+    DataListFilters,
+  );
+
   return (
-    <DataListContext.Provider value={props}>
-      <InternalDataList {...props} />
+    <DataListContext.Provider
+      value={{ searchComponent, filterComponent, ...props }}
+    >
+      <InternalDataList />
     </DataListContext.Provider>
   );
 }
 
-function InternalDataList<T extends DataListObject>({
-  data,
-  headers,
-  loading = false,
-  filterApplied = false,
-  children,
-  title,
-  totalCount,
-  headerVisibility = { xs: true, sm: true, md: true, lg: true, xl: true },
-}: DataListProps<T>) {
-  const allLayouts = getCompoundComponents<DataListLayoutProps<T>>(
+function InternalDataList() {
+  const {
+    data,
+    headers,
+    loading = false,
+    filterApplied = false,
```

Comment:
No action needed at the moment, but do we still need this `filterApplied` prop now that the filters compound component has been added

Reply:
Gonna reevaluate on the other ticket about filtering and empty state

Reply:
@darryltec sounds good :)